bindings/rust/s2n-tls/src/client_hello.rs

```diff
@@ -324,28 +315,23 @@ pub mod fingerprint {
                 .unwrap()
                 .build()
                 .unwrap();
-            let pair = crate::testing::tls_pair(config);
+            let mut pair = TestPair::from_config(&config);
 
             // client_hellos can not be accessed before the handshake
-            assert!(pair.client.0.connection().client_hello().is_err());
-            assert!(pair.server.0.connection().client_hello().is_err());
-
-            let pair = poll_tls_pair(pair);
-            let server_conn = pair.server.0.connection();
-            let client_conn = pair.server.0.connection();
```

Comment:
There was a copy-paste mistake in this test. The `client_hello` method is not available for client connections, which is a property of the underlying C implementation. This never would have worked, and the client assertion is removed in the new version.